File: android/app/src/main/java/com/mengzhen/app/receiver/BootReceiver.java
package com.mengzhen.app.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.mengzhen.app.scheduler.AlarmScheduler;

/**
 * 设备重启后恢复所有闹钟
 */
public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            AlarmScheduler scheduler = AlarmScheduler.getInstance(context);
            scheduler.restoreAllAlarms();
        }
    }
}

File: android/app/src/main/java/com/mengzhen/app/receiver/BootReceiver.kt
package com.mengzhen.app.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.mengzhen.app.scheduler.AlarmScheduler

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action) {
            Intent.ACTION_BOOT_COMPLETED,
            Intent.ACTION_MY_PACKAGE_REPLACED,
            Intent.ACTION_TIME_CHANGED,
            Intent.ACTION_TIMEZONE_CHANGED,
            android.app.AlarmManager.ACTION_SCHEDULE_EXACT_ALARM_PERMISSION_STATE_CHANGED,
            "android.intent.action.QUICKBOOT_POWERON",
            "com.htc.intent.action.QUICKBOOT_POWERON" -> {
                // Android 15+ 禁止从开机/系统广播直接启动 mediaPlayback FGS。
                // 先恢复 AlarmManager；真正播放由随后的精确闹钟触发。
                AlarmScheduler.get(context).restoreAllAlarms(deferPlaybackToAlarm = true)
            }
        }
    }
}
